doit/cmd_base.py

```diff
@@ -446,8 +480,17 @@ def execute(self, params, args):
         :param params: instance of cmdparse.DefaultUpdate
         :param args: list of string arguments (containing task names)
         """
-        self.task_list, dodo_config = self.loader.load_tasks(
-            self, params, args)
+
+        # distinguish legacy and new-style task loader API when loading tasks:
+        if isinstance(self.loader, TaskLoader2):
```

Comment:
I would prefer we add an ``API=2`` check, so users are not forced to be subclass ``TaskLoader2`` as long as they implement the interface...